lib/HomeSeerSystemObject.js

```diff
@@ -219,8 +219,8 @@ class HomeSeerSystem
 					&& 	(this.HomeSeerDevices[reference].status.device_type.Device_Type === 999) )
 					return true;		
 		// Thermostat
-		if ( 		(this.HomeSeerDevices[reference]?.status.device_type.Device_API === 16)
-					&& 	(this.HomeSeerDevices[reference].status.device_type.Device_Type === 99)
+		if ( 		(this.HomeSeerDevices[reference]?.status.device_type.Device_API === 16 || this.HomeSeerDevices[reference]?.status.device_type.Device_API === 512) \\new HS4 plugin has a API of 512 and device type of 8
+					&& 	(this.HomeSeerDevices[reference].status.device_type.Device_Type === 99 || this.HomeSeerDevices[reference].status.device_type.Device_Type === 8)

```

Comment:
I think this is cleaner if you leave the original Thermostat code as it was and just add a new block for the Nest. I.e., like . . .

		// Thermostat
		if ( 		(this.HomeSeerDevices[reference]?.status.device_type.Device_API === 16)
					&& 	(this.HomeSeerDevices[reference].status.device_type.Device_Type === 99)
					&&	(this.HomeSeerDevices[reference].status.device_type.Device_SubType === 0) )
					return true;
		// Detect New NEST Thermostat API
		if ( 		(this.HomeSeerDevices[reference]?.status.device_type.Device_API === 512)
					&& 	(this.HomeSeerDevices[reference].status.device_type.Device_Type === 8)
					&&	(this.HomeSeerDevices[reference].status.device_type.Device_SubType === 0) )
					return true;

Reply:
Looks good, made the change.